gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/specgen/usecase/SpecGenRequestUseCaseTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.specgen.usecase;
+
+import static io.gravitee.apim.core.specgen.model.ApiSpecGenOperation.GET_STATE;
+import static io.gravitee.apim.core.specgen.model.ApiSpecGenOperation.POST_JOB;
+import static io.gravitee.apim.core.specgen.model.ApiSpecGenRequestState.UNAVAILABLE;
+import static io.gravitee.rest.api.service.common.UuidString.generateRandom;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import io.gravitee.apim.core.specgen.model.ApiSpecGen;
+import io.gravitee.apim.core.specgen.model.ApiSpecGenRequestReply;
+import io.gravitee.apim.core.specgen.model.ApiSpecGenRequestState;
+import io.gravitee.apim.core.specgen.query_service.ApiSpecGenQueryService;
+import io.gravitee.apim.core.specgen.service_provider.SpecGenProvider;
+import io.gravitee.apim.core.specgen.use_case.SpecGenRequestUseCase;
+import io.gravitee.definition.model.v4.ApiType;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@TestInstance(Lifecycle.PER_CLASS)
+@ExtendWith(MockitoExtension.class)
+public class SpecGenRequestUseCaseTest {
+
+    @Mock
+    ApiSpecGenQueryService apiSpecGenQueryService;
```

Comment:
Same here with a ApiSpecGenQueryServiceInMemmory 🥺 
This would help maintain consistency in the way tests are run here. Mocks are avoided whenever possible for UseCase tests. This allows us to have high-level tests that better reflect business cases. 

no `when` you initialize your api once in BeforEach and you'll have it for all your tests. simple and more realistic no ? 

If you had used the ApiCrudService the InMemory already exist